Sources/MapboxCoreNavigation/EHorizon/RoadObjectsStore.swift

```diff
@@ -56,6 +56,36 @@ public final class RoadObjectsStore {
         return native.getRoadObjectIdsByEdgeIds(forEdgeIds: edgeIdentifiers.map(NSNumber.init))
     }
 
+    /**
+     Adds a road object to be tracked in the electronic horizon. In case if an object with such identifier already exists, updates it.
+     NB: road objects obtained from route alerts cannot be added via this API.
+
+     - parameter roadObject: Custom road object, acquired from `RoadObjectMathcer`.
+     */
+    public func addCustomRoadObject(_ roadObject: RoadObject) {
+        guard let nativeObject = roadObject.native else {
+            assertionFailure("You can only add matched custom road objects, acquired from RoadObjectMatcher.")
```

Comment:
It might be better to make the method `throwing` and throw an error here. Similar to what Maps do for custom style layers/sources

Reply:
Also, I think `preconditionFailure(_:)` would be more specific in this case.

Reply:
I used `assertionFailure` intentionally not to crash in Release configurations and identify that such state is non-fatal.
`preconditionFailure` is about non-recoverable failures.

Reply:
Chose `preconditionFailure` as the case is only possible if the user passes `RoadObject` created with public init, which is a programming error, not a state that should or can be handled.